src/Contexts/data-context.js

```diff
@@ -0,0 +1,184 @@
+import {useContext, createContext, useReducer, useEffect, useState } from "react";
+import axios from "axios";
+
+const DataContext = createContext();
+
+export function DataProvider({children}) {
+
+    console.log("Data context called")
+    
+    const [ data, setData ] = useState([]);
+    const [ loading, setLoading ] = useState(true);
+  
+    useEffect(() => {
+        (async function () {
+            
+            console.log("data called")
+            const response = await axios('https://nftBaazarAPI.rawheatg.repl.co/test');
+            console.log("Data from useEfect: ", data)
+            setData(response.data.data);
+            
+            setLoading(false);
+            console.log("Data from useEfect: ", data)
+        }) ()
+    }, []);
```

Comment:
You can set your cart data and wishlist data along with product data from backend.